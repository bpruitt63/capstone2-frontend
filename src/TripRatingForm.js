import React from 'react';
import { Button, Form, Label, Input, InputGroup, InputGroupAddon, InputGroupText } from 'reactstrap';
import './static/styles/Trips.css';
import {useHandleChange, useErrors} from './hooks';
import Errors from './Errors';
import BoateyApi from './BoateyApi';

/** Form for rating a trip.  1 to 5, whole number ratings */
function TripRatingForm({trip, username, setRating, isMobile}) {

    const initialState = trip.tripRating ? 
                            {rating: trip.tripRating} : {rating: 3};
    const [data, handleChange] = useHandleChange(initialState);
    const [apiErrors, getApiErrors, setApiErrors] = useErrors();

    const handleSubmit = async (e) => {
        e.preventDefault();
        try{
            const rating = await BoateyApi.rateTrip(username, trip.tripId, data);
            setRating(rating.tripRating);
            setApiErrors({});
        } catch (err) {
            getApiErrors(err);
        };
    };

    return (
        <div className='tripRatingContainer'>
        <div className='tripRatingForm'>
            <Errors apiErrors={apiErrors} />
            <Form onSubmit={handleSubmit}>
                {!isMobile &&
                    <InputGroup>
                        <InputGroupAddon addonType="prepend">
                            <InputGroupText className='rate'>
                                <Label htmlFor={`${trip.tripId}rating`}>Rate This Trip: </Label>
                            </InputGroupText>
                        </InputGroupAddon>
                        <Input type='select' 
                                name='rating'
                                id={`${trip.tripId}rating`}
                                value={data.rating}
                                className='ratingInput'
                                onChange={handleChange}>
                            <option value='1'>1</option>
                            <option value='2'>2</option>
                            <option value='3'>3</option>
                            <option value='4'>4</option>
                            <option value='5'>5</option>
                        </Input>
                        <InputGroupAddon addonType="append" >
                            <Button>Submit Rating</Button>
                        </InputGroupAddon>
                    </InputGroup>
                }
                {isMobile && 
                    <>
                        <Label htmlFor={`${trip.tripId}MRating`} className='detailTitle'>
                            Rate This Trip
                        </Label>
                        <br/>
                        <Input type='select' 
                                name='rating'
                                id={`${trip.tripId}MRating`}
                                className='mRatingInput'
                                value={data.rating}
                                onChange={handleChange}>
                            <option value='1'>1</option>
                            <option value='2'>2</option>
                            <option value='3'>3</option>
                            <option value='4'>4</option>
                            <option value='5'>5</option>
                        </Input>
                        <br/>
                        <Button>Submit Rating</Button>
                    </>
                }
            </Form>
        </div>
        </div>
    );
};

export default TripRatingForm;