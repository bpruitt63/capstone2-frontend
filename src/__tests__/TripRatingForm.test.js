import React from 'react';
import { render, screen } from '@testing-library/react';
import TripRatingForm from '../TripRatingForm';

test('renders', () => {
    render(<TripRatingForm trip={{tripRating: 3}}/>);
    expect(screen.getByText(/Rate This Trip:/)).toBeInTheDocument;
});

test('matches snapshot', () => {
    const {asFragment} = render(<TripRatingForm trip={{tripRating: 3}}/>);
    expect(asFragment()).toMatchSnapshot();
});