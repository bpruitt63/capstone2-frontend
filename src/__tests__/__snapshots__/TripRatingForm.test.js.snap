// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tripRatingContainer"
  >
    <div
      class="tripRatingForm"
    >
      <div
        class="Errors"
      />
      <form
        class=""
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="rate input-group-text"
            >
              <label
                class=""
                for="undefinedrating"
              >
                Rate This Trip: 
              </label>
            </span>
          </div>
          <select
            class="ratingInput form-control"
            id="undefinedrating"
            name="rating"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
          </select>
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-secondary"
            >
              Submit Rating
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
